### PR TITLE
add preview image to share template page

### PR DESCRIPTION
**Pull request by STY1997**
As the court screenshot feature has already been completed, the court preview image has been added to the create template page and image link can be stored in the database. 
Screen Shot 2022-10-26 at 21 24 53

### CODE REVIEW

File: src/components/CreateTemplate/index.tsx [hunk 1]
```diff
@@ -90,12 +108,19 @@ function CreateTemplate({ isOpen, onClose }: Props) {
       }));
       return;
     }
+    if (!courtDataUrl) {
+      const title = `Fail to get courtDataUrl`;
```
Comment (Double-Twelve) [inline src/components/CreateTemplate/index.tsx:112]: Customers may not know what courtDataUrl is, maybe it's better to change it to "Fail to get court image"

File: src/interfaces/design.d.ts [hunk 2]
```diff
@@ -43,6 +43,7 @@ export interface ICourtColor {
 }
 
 export interface ISaveDesign {
+  designer?: string;
```
Comment (Double-Twelve) [inline src/interfaces/design.d.ts:46]: what is case that _designer_ is undefined?

File: src/utils/manageExternalImage.ts [hunk 3]
```diff
@@ -1,3 +1,9 @@
+export const generateImageFromDataUrl = async (courtDataUrl: string) => {
+  const base64 = await fetch(courtDataUrl);
+  const blob = await base64.blob();
+  return new File([blob], "default", { type: "image/png" });
+};
```
Comment (Double-Twelve) [inline src/utils/manageExternalImage.ts:5]: does this method need to export?

File: src/interfaces/design.d.ts (hunk 2 above)
Comment (STY1997) [reply src/interfaces/design.d.ts:46]: > what is case that _designer_ is undefined?

This interface will be called in both creating template and generating shopping cart item. And the designer field will be only used to create the new template.

**Pull request closed without merging**